### ISSUE TITLE
Feature Request: recurring maintenance windows for services

### ISSUE THREAD

**Issue by ryandzink** [terraform-providers/terraform-provider-pagerduty#123]
PagerDuty has the ability to schedule recurring maintenance windows through their API, although this is not yet available through the UI. This would be invaluable for setting up scheduled maintenance (e.g. service restarts, server reboots) without going through a separate process after deploying infrastructure to create the recurring maintenance windows via API call.

[Reference](https://support.pagerduty.com/docs/maintenance-windows#section-recurring-maintenance-mode-for-services)

### Terraform Version
0.11.11

### Affected Resource(s)
pagerduty_maintenance_window

**Comment by heimweh** [comment]
Hi @ryandzink ,

Thanks for opening this issue. 

Correct me if I’m wrong but I think this is kind of possible already using the current `pagerduty_maintenance_window` resource. 
This is accomplished by creating multiple maintenance windows basically.

Does that help? 🙂

Best,

Alexander

**Comment by ryandzink** [comment]
Hey @heimweh ,

That is close to what I'm trying to do, but it just seems clunky to have to do (for example) 52 different maintenance windows for a weekly maintenance for the next year. However, looking deeper at the python script that I referred to, there is no open-ended "recurring maintenance window" in pagerduty, and it's just creating multiple maintenance windows rather than what I thought it was doing. Until Pagerduty supports that, my request doesn't make sense, so I'll go ahead and close it.

I've settled on our maintenance activities making calls to the Pagerduty API and manually creating maintenance windows per maintenance, which seems to be working well.
